front/src/components/Table/index.tsx

```diff
@@ -0,0 +1,110 @@
+import React from "react";
+import { useSelector } from "react-redux";
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  Paper,
+} from "@material-ui/core";
+
+import { selectBooksBasket } from "features/books/booksSlice";
+
+import { useStyles, InfoElement } from "./styles";
+
+function createData(
+  name: string,
+  calories: number,
+  fat: number,
+  carbs: number,
+  protein: number
+) {
+  return { name, calories, fat, carbs, protein };
+}
+
+interface Book {
+  author: string;
+  cover_url: string;
+  currency: string;
+  id: number;
+  pages: number;
+  price: number;
+  title: string;
+  value: number;
+}
+
+export default function BasicTable() {
+  const basket = useSelector(selectBooksBasket);
+  console.log(basket);
+
+  const classes = useStyles();
+
+  const newRows = basket.map((book: Book) => {
+    const { author, title, pages, price, currency, cover_url, id } = book;
+    return {
+      author,
+      title,
+      pages,
+      price,
+      currency,
+      cover_url,
+      id,
+    };
+  });
+
+  const rows = [
+    createData("Frozen yoghurt", 159, 6.0, 24, 4.0),
```

Comment:
takie coś jest nieładne, bo zamienisz miejscami jeden parametr z innym, albo jakiś dodasz jako np. drugi, to musisz wszędzie to pozamieniać. 
Takie struktury to najlepiej w osobnym data.ts jako:

const deserts: IDesert[] = [
{
"name": "Frozen yoghurt,
"calories": 159,
"fat": 6,
"carbs": 24,
"protein": 4,
},
...
]

.0 nie musisz dodawać, js domyślnie robi liczby float. Czasem możesz je castować na integera. Typescript uznaje po prostu "number".
Tak też to jest z reguły trzymane w bazie i zwracane.

I w pliku interfaces.ts albo models.ts:
type IDesert = {
name: string,
  calories: number,
  fat: number,
  carbs: number,
  protein: number
}

nie potrzebujesz tego createData w ogóle

Reply:
Done.